packages/vitest/src/node/core.ts

```diff
@@ -386,9 +386,13 @@ export class Vitest {
   }
 
   private async getTestDependencies(filepath: WorkspaceSpec, deps = new Set<string>()) {
+    const [, workspaceFilepath] = filepath
     const addImports = async ([project, filepath]: WorkspaceSpec) => {
       if (deps.has(filepath))
         return
+      if (workspaceFilepath !== filepath)
+        deps.add(filepath)
```

Comment:
nit: this is probably just a minor thing, but I think we could just accumulating everything but filter `workspaceFilepath` later? something like:

```tsx
  const addImports = async (...) => {
      if (deps.has(filepath))
        return
      deps.add(filepath)
      ...
  } 

  await addImports(filepath)
  deps.delete(filepath[1])
```